src/iroh.rs

```diff
@@ -471,6 +471,51 @@ impl VeilidIrohBlobs {
         Ok(collection.keys().cloned().collect())
     }
 
+    pub async fn list_files_from_hash(&self, collection_hash: &Hash) ->  Result<Vec<String>> {
+        // Fetch the collection data using the collection hash
+        let entry = self
+        .store
+        .get(collection_hash)
+        .await?
+        .ok_or_else(|| anyhow!("Collection not found for hash: {}", collection_hash))?;
+        
+        // Read the serialized collection data directly
+        let collection_data = self.read_bytes(*collection_hash).await?;
+
+        // Deserialize the collection into a HashMap
+        let collection: HashMap<String, Hash> = from_slice(&collection_data)
+            .map_err(|err| anyhow!("Failed to deserialize collection: {:?}", err))?;
+
+        // Return the list of file names (keys in the HashMap)
+        Ok(collection.keys().cloned().collect())
+    }
+
+    pub async fn get_file_from_collection_hash(
+        &self,
+        collection_hash: &Hash,
+        path: &str,
+    ) -> Result<Hash> {
+        // Fetch the collection entry from the store using the collection hash
+        let entry = self
```

Comment:
these should be put in a method like `get_collection_from_hash to reuse. get_collection should also use this new method

Reply:
7e9cffd7c67b15c7eade7fd003136eca28ff7bfc
Refactors the list_files_from_hash method. It removes unnecessary code and adds a new method called get_collection_from_hash. The get_collection_from_hash method retrieves the collection HashMap using the collection hash.

Reply:
get_collection currently gets the collection from the name.

if it uses get_collection_from_hash, the following methods will need to be refactored to take the hash instead of the name:
- set_file
- get_file
- list_files

should i refactor to make these use the collection_hash instead of collection_name?


current implementation:
`set_file(collectionName, path: string, hash) -> Result<Hash>` (return collection hash)
`get_file(collectionName, path) -> Result<Hash>`
`delete_file(collectionNAme, path) -> Result<Hash> `(return collection hash)
`list_files(collectionNAme)  -> Result`<Vec<string>>